bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/ManagedThingProvider.java

```diff
@@ -31,11 +40,21 @@
  */
 @NonNullByDefault
 @Component(immediate = true, service = { ThingProvider.class, ManagedThingProvider.class })
-public class ManagedThingProvider extends DefaultAbstractManagedProvider<Thing, ThingUID> implements ThingProvider {
+public class ManagedThingProvider extends AbstractManagedProvider<Thing, ThingUID, ThingDTO> implements ThingProvider {
+
+    private final ThingTypeRegistry thingTypeRegistry;
+    private final ThingManager thingManager;
+
+    private final Map<String, Thing> things;
```

Comment:
Does it really make sense to store a reference for all managed things here? Wouldn't it be better to move the check for duplicates to the `ThingManager` or use the `ThingRegistry`.